### PR TITLE
SIGTERM will gracefully shutdown the framework.

### LINKED ISSUE #462: installing a framework after teardown
Using Marathon I removed ES Scheduler, but this left Executors running. In order to remove them I teared down the framework by `curl -X POST -d 'frameworkId=XXXXXXXX-b036-4cb7-af53-4c837dc9521d-0002' http://${MASTER_IP}:5050/master/teardown`. This successfully removed all the framework tasks - executors. 

However now Mesos Cluster rejects my attempts to re-install the framework. 
```
I0115 12:54:57.916470 28856 sched.cpp:1024] Got error 'Framework has been removed'
I0115 12:54:57.916509 28856 sched.cpp:1805] Asked to abort the driver
I0115 12:54:57.916824 28856 sched.cpp:1070] Aborting framework '8ca5c18f-b036-4cb7-af53-4c837dc9521d-0001'
```

Below I copy an answer from `user@mesos.apache.org`. Is it possible the scheduler gets the old ID from Zookeeper? What changes should we introduce to enable re-installation of the framework?

### PR DESCRIPTION
**Pull request by philwinder**
All executors will be killed.
Zk state is removed.

To test:
```
$ docker kill --signal=SIGTERM ${SCHEDULER_ID}
$ docker exec -it ${ZOOKEEPER_ID} bash
# ./bin/zkCli.sh
[zk: localhost:2181(CONNECTED) 1] ls /elasticsearch/mesos-ha
[]
```
Confirm no zk nodes.
Fixes #462

### CODE REVIEW

**Comment by philwinder** [comment]
Jenkins pass: https://ci.infra.container-solutions.com/view/All/job/ElasticsearchPullRequests/286/console

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/state/StatePath.java [hunk 1]
```diff
@@ -43,4 +43,9 @@ public Boolean exists(String key) throws IOException {
         }
         return exists;
     }
+
+    public void rm(String key) throws IOException {
```
Comment (sadovnikov) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/state/StatePath.java]: this method deserves a description

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/state/StatePath.java [hunk 2]
```diff
@@ -43,4 +43,9 @@ public Boolean exists(String key) throws IOException {
...
+        key = key.replace(" ", "");
```
Comment (sadovnikov) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/state/StatePath.java]: Where would these " " come from? Without explanation it looks like a manipulation of a key to remove

Comment (philwinder) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/state/StatePath.java]: Fair point. Not sure. Another method had it. I obviously had problems with spaces in the past. Will remove.

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/state/StatePath.java (hunk 1 above)
Comment (philwinder) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/state/StatePath.java]: It should be private. And I could probably inline now after removing the replace. And everyone understands what `rm` does, so a description would be a waste of lines of code.

**Pull request merged**